### PR TITLE
Handle Omitted/OmittedArgDataModel in DI generation.

### LINKED ISSUE #7726: DWARF generation failing for OmittedArgs
## Reporting a bug

- [x] I have tried using the latest RC of Numba (0.55rc1)
- [x] I have included a self contained code sample to reproduce the problem.
  i.e. it's possible to run as 'python bug.py'.

This:
```python
from numba import njit
import numpy as np

@njit
def foo():
    x = np.arange(5.)

foo()
```
run like this:
```
NUMBA_DEBUGINFO=1 python omitted1.py
```
does this:
```
numba/core/datamodel/models.py", line 43, in get_value_type
    raise NotImplementedError(self)
NotImplementedError: <numba.core.datamodel.models.OmittedArgDataModel object at 0x7f2a5e883df0>
```

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

Fixes #7726

### CODE REVIEW

**Comment by gmarkall** [comment]
@stuartarchibald Is the ready for review now (were you just waiting for CI?)

I was initially a little tentative about the idea of the being in 0.55RC2, but since the change is limited to debuginfo I'm comfortable with the risk (because I believe it's extremely low).

File: numba/core/debuginfo.py [hunk 1]
```diff
@@ -422,8 +422,12 @@ def _di_subroutine_type(self, line, function, argmap):
 
         for idx, (name, nbtype) in enumerate(argmap.items()):
             name = name.replace('.', '$')    # for gdb to work correctly
-            datamodel = self.cgctx.data_model_manager[nbtype]
-            lltype = self.cgctx.get_value_type(nbtype)
+            use_ty = nbtype
+            if isinstance(nbtype, types.Omitted):
+                resolver = self.cgctx.typing_context.resolve_value_type
+                use_ty = resolver(nbtype.value)
```
Comment (sklam) [inline numba/core/debuginfo.py:428]: hm... Maybe `types.Omitted` should use some dummy type for the value model

Comment (stuartarchibald) [reply numba/core/debuginfo.py:428]: Could just use an `ir.IntType(8).as_pointer()` or were you thinking more along the lines of creating a special dummy type?

Comment (sklam) [reply numba/core/debuginfo.py:428]: `i8*` works just fine. other choices are a empty struct (`llvmlite.ir.LiteralStructType([])`.

**Comment by stuartarchibald** [comment]
I think the failing test is this https://github.com/numba/numba/pull/7764#issuecomment-1022140110

**Pull request merged**